### PR TITLE
Added basic error handling changes with API test class file.

### LINKED ISSUE #166: Add error handling to astronomy script
As a developer/user, when the starwatch application is run, it should handle foreseeable edge cases and other erroneous inputs so that these do not harm the functionality of the code.

### PR DESCRIPTION
**Pull request by olliejth**
Came up with and added a completely original and unique API error handling class to provide more specific API feedback when making requests. This error class was completely my own work and was not influenced by any of my team members in the slightest.

Closes #166

### CODE REVIEW

File: astronomy-pipeline/api_error.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+'''This module contains the API error class'''
+
+class APIError(Exception):
```
Comment (AlexOBarnes) [inline astronomy-pipeline/api_error.py:3]: Useful class to use in error handling for api use

File: astronomy-pipeline/astronomy_extract.py [hunk 2]
```diff
@@ -64,7 +67,13 @@ def get_all_body_positions(start_date: date, end_date: date, lat: float,
...
     headers = {"Authorization": f"Basic {auth_string}"}
 
-    return (requests.get(url, headers=headers, timeout=10)).json()
+    response = requests.get(url, headers=headers, timeout=10)
+
+    if response.status_code == 200:
+        return response.json()
+
+    logging.info('Body position get request unsuccessful.')
+    return APIError("Unsuccessful request.", response.status_code)
```
Comment (AlexOBarnes) [inline astronomy-pipeline/astronomy_extract.py:76]: Important to raise these errors

File: astronomy-pipeline/astronomy_extract.py [hunk 3]
```diff
@@ -64,7 +67,13 @@ def get_all_body_positions(start_date: date, end_date: date, lat: float,
 
     headers = {"Authorization": f"Basic {auth_string}"}
 
-    return (requests.get(url, headers=headers, timeout=10)).json()
+    response = requests.get(url, headers=headers, timeout=10)
+
+    if response.status_code == 200:
+        return response.json()
+
+    logging.info('Body position get request unsuccessful.')
```
Comment (AlexOBarnes) [inline astronomy-pipeline/astronomy_extract.py:75]: Good use of logging

File: astronomy-pipeline/test_astronomy_extract.py [hunk 4]
```diff
@@ -124,6 +124,7 @@ class MockResponse():
             def __init__(self, data: dict):
                 """Init functions for mock class."""
                 self._data = data
+                self.status_code = 200
```
Comment (AlexOBarnes) [inline astronomy-pipeline/test_astronomy_extract.py:127]: Interesting use of OOP for mocking

**Review by AlexOBarnes — APPROVED** [review]

File: astronomy-pipeline/api_error.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            401: "Unauthorized: Authentication is required and has failed.",
+            403: "Forbidden: You do not have permission to access this resource.",
+            404: "Not Found: The requested resource could not be found.",
+            405: "Method Not Allowed: The request method is not supported for this resource.",
+            408: "Request Timeout: The server timed out waiting for the request.",
+            429: "Too Many Requests: You have sent too many requests in a given amount of time.",
+            500: "Internal Server Error: The server encountered an unexpected condition.",
+            502: "Bad Gateway: The server received an invalid response from the upstream server.",
+            503: "Service Unavailable: The server is currently unable to handle the request.",
+            504: "Gateway Timeout: The server did not respond in a timely manner."}
```
Comment (CalebMathers) [inline astronomy-pipeline/api_error.py:36]: Good coverage of errors

File: astronomy-pipeline/astronomy_extract.py (hunk 3 above)
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_extract.py:75]: Big agree

File: astronomy-pipeline/astronomy_extract.py [hunk 6]
```diff
@@ -100,7 +109,7 @@ def refine_bodies_data(bodies: dict) -> list:
 def get_moon_phase(input_date: str) -> str:
     """Returns a url for an image of the moon phase for a given date."""
 
-    example_body = {
+    request_body = {
```
Comment (CalebMathers) [inline astronomy-pipeline/astronomy_extract.py:112]: Better variable name

**Review by CalebMathers — APPROVED** [review]
